test/format/config.c

```diff
@@ -989,18 +996,32 @@ config_mirrors(void)
     explicit_mirror = config_explicit(NULL, "runs.mirror");
     if (!explicit_mirror && mmrand(NULL, 1, 10) < 9)
         return;
-    config_off_all("runs.mirror");
 
     /*
      * We can't mirror if we don't have enough tables. A FLCS table can be a mirror, but it can't be
      * the source of the bulk-load mirror records. Find the first table we can use as a base.
      */
     for (i = 1; i <= ntables; ++i)
-        if (tables[i]->type != FIX)
+        if (tables[i]->type != FIX && !NT_EXPLICIT_OFF(tables[i], RUNS_MIRROR))
             break;
-    if (ntables < 2 || i > ntables) {
-        if (explicit_mirror)
-            WARN("%s", "table selection didn't support mirroring, turning off mirroring");
+
+    if (i > ntables && explicit_mirror) {
+        WARN("%s", "table selection didn't support mirroring, turning off mirroring");
```

Comment:
Since you removed the line above `config_off_all("runs.mirror");` is this line (and other similar ones below) true anymore before returning?

Reply:
That's a good question, I could have been clearer about that. It's necessary since it deletes all of the info about how mirroring was initially configured, so if I leave it in, none of the subsequent `NT_EXPLICIT_OFF` macros will work.

As for it being OK, `explicit_mirror` was originally being set before the `config_off_all` call, so the value of that hasn't changed. The check you've commented on here (and the analogous one below) still work as intended, since there are two ways `explicit_mirror` can be set:
1. table(s) set it explicitly on, in which case the code works as before
2. table(s) set it explicitly off, in which case we'll skip the "randomisation" step above (correctly IMO), and we'll still do this specific check you've commented on, which works fine because there's now a `NT_EXPLICIT_OFF` check. The same argument applies to the "available tables" check below.

Reply:
I think the confusing part for me then is the "turning off mirroring" implies some action and all we're doing here is returning. Maybe a comment here about where it is being turned off, or change the wording somehow.

Reply:
Ah I see now. Agreed that's a bit confusing - I've added `config_off_all` calls so it's clearer what we're intending. It's a little unnecessary right now since nothing looks at the mirroring config after we've configured it and the table state is "correct" at the time of creation. But that may not be true forever.

I also fixed a segfault in an edge case where we only have FLCS tables and nothing has explicitly configured mirroring - for these "do we have enough tables" checks, we want to warn when `explicit_mirror` is set, but bail on configuration regardless.